System/Process.hs

```diff
@@ -616,6 +613,7 @@ waitForProcess ph@(ProcessHandle _ delegating_ctlc) = do
 #else
         return $ ExitFailure (-1)
 #endif
+  where lockWaitpid m = withMVar (waitpidLock ph) $ \() -> m
```

Comment:
Let's add a comment here to explain the problem and the solution, and point to this ticket.